### PR TITLE
Clean up Redis transport layer

### PR DESCRIPTION
**Pull request by penberg**
The Redis transport layer seems to have originated as a copy-paste of 
the CQL transport layer. This pull request removes bunch of unused
and commented out bits of code, and also does some minor cleanups
like organizing includes, to make the code more readable.

### CODE REVIEW

File: redis/server.cc [hunk 1]
```diff
@@ -41,7 +41,7 @@ namespace redis_transport {
 
 static logging::logger logging("redis_server");
 
-redis_server::redis_server(distributed<service::storage_proxy>& proxy, distributed<redis::query_processor>& qp, auth::service& auth_service, redis_server_config config)
+redis_server::redis_server(seastar::sharded<service::storage_proxy>& proxy, seastar::sharded<redis::query_processor>& qp, auth::service& auth_service, redis_server_config config)
```
Comment (nyh) [inline redis/server.cc:48]: I don't mind this change too much, but why did you have to write seastar::sharded and not just sharded?

File: redis/server.hh [hunk 2]
```diff
@@ -10,7 +10,7 @@
  * the Free Software Foundation, either version 3 of the License, or
  * (at your option) any later version.
  *
- * Scylla is distributed in the hope that it will be useful,
+ * Scylla is seastar::sharded in the hope that it will be useful,
```
Comment (nyh) [inline redis/server.hh:13]: Ooops :-) This will have to be fixed back :-)

**Review by nyh — COMMENTED** [review]
Looks good, except one amusing search-and-replaco in a comment :-)

File: redis/server.hh (hunk 2 above)
Comment (avikivity) [reply redis/server.hh:13]: Yes please. Although the comment is still correct.

File: redis/server.cc (hunk 1 above)
Comment (penberg) [reply redis/server.cc:48]: I was not aware there's a preference. I just switched to the fully qualified name in the header and source file for consistency.

**Pull request merged**